### PR TITLE
Add UndocumentedPublicProperty rule - closes #1670

### LINKED ISSUE #1670: UndocumentedPublicFunction should also report properties
## Expected Behavior
A property without documentation should also be reported

## Observed Behavior
no error (only function was reported)

## Steps to Reproduce
https://sonarcloud.io/project/issues?id=tegonal_atrium-ktor&open=AWsMs313vwsQMbwCGitv&resolved=false&types=CODE_SMELL

Error is only reported for the function as you can see.
To reproduce it manually:
1. git clone https://github.com/tegonal/atrium-ktor
2. ./gr detekt

## Your Environment
* Version of detekt used: 1.0.0-RC14
* Version of Gradle used (if applicable): 5.4.1
* Link to your project (if it's a public repository): https://github.com/tegonal/atrium-ktor

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+					var a = x
+				}
+			"""
+            assertThat(subject.compileAndLint(code)).isEmpty()
+        }
+
+        it("does not report public properties in internal class") {
+            val code = """
+    			internal class NoComments {
+                    public val a = 1
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: Formatting looks off in the test cases on GitHub.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    				}
+    			}
+			"""
+            assertThat(subject.compileAndLint(code)).hasSize(2)
+        }
+
+        it("does not report documented public property") {
+            val code = """
+    			/**
+			     * Comment
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: Formatting looks off here

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt [hunk 3]
```diff
@@ -81,12 +76,8 @@ class UndocumentedPublicFunctionSpec : Spek({
         it("does not report public functions in internal class") {
             val code = """
     			internal class NoComments {
-
-					fun nope0() {}
-					public fun nope1() {}
-					internal fun nope2() {}
-					protected fun nope3() {}
-					private fun nope4() {}
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt]: I would leave the different visibility kinds inside the testcase

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt (hunk 1 above)
Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicPropertySpec.kt]: I replaced all tabs with spaces in test snippet code.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt (hunk 3 above)
Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/UndocumentedPublicFunctionSpec.kt]: I put them back.

**Comment by arturbosch** [comment]
Unfortunately I messed it up with the GitHub gui to fix the conflicts :(

**Pull request merged**